bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseThingHandlerFactory.java

```diff
@@ -98,7 +101,11 @@ public void unregisterHandler(Thing thing) {
         ServiceRegistration<ThingHandler> thingHandlerServiceRegistration = thingHandlers
                 .remove(thing.getUID().toString());
         if (thingHandlerServiceRegistration != null) {
+            logger.debug("Removing service registration for handler of thing " + thing.getUID().getAsString());
```

Comment:
Don't build / concat strings in the logger arguments, this should be done by the logger itself.
Use '{}' and the arg list.